src/Model/Contact.php

```diff
@@ -3529,7 +3529,15 @@ public static function searchByName(string $search, string $mode = '', int $uid
 			$networks[] = Protocol::OSTATUS;
 		}
 
-		$condition = ['network' => $networks, 'failed' => false, 'deleted' => false, 'uid' => $uid];
+		$condition = [
+			'network'        => $networks,
+			'server-failed'  => false,
+			'server-blocked' => false,
```

Comment:
The problem with this condition is that local users won't know why a user search returns fewer results than they expect even though it depends on an action by their local admin (all other conditions are fine). The check on this field must be done later with an explanatory label like I did in https://github.com/friendica/friendica/pull/12954

Reply:
I don't like this ambiguous message. I think we either had to not show results from blocked servers or we had to display them with the hint that this account is on a blocked server. But some "your search results contained some results on servers that we block" doesn't help, since you don't know if this was the result you searched for.

Reply:
I'm fine with the latter. More information is always better.

Reply:
Then I guess that we need a user setting for that. When we look into the issue, then we see that the request was for not displaying the search results from blocked servers at all.

Reply:
Please don't add a user setting for this. Let's display the result with no interaction possible and an explanation.

Reply:
If this is about a Mastodon API endpoint, the SQL condition makes sense, but for anything Friendica-specific, I'd prefer the `server-blocked` flag to just be part of the result contact record.